site/content/entry/yuzu-mini-dev-3/index.md

```diff
@@ -0,0 +1,102 @@
+
+**Q: So you decided that it was time to take matters into your own hands and add support yourself?**
+
+**E:** Knowing that other emulators (notably [Dolphin](https://dolphin-emu.org/)) had proper support for the official GameCube adapter, 
+and considering I was looking to sharpen my programming skillset, 
+I decided to take the opportunity and see if I can bring that support over to yuzu, opening the option to use the GameCube controller for anyone on the emulator.
+
+**Q: I'm sure our GC controller users are very happy with your contribution! Why did you choose to tackle such a large task as NVDEC?**
+
+**E:** I found the code review for the GC adapter implementation to be really useful. 
+I wanted to learn more from the very sharp yuzu developers and continued to make more contributions and became more familiar with the yuzu dev team. 
+One feature that I saw many yuzu fans request was for video decoding. 
+I learned that [ogniK](https://github.com/ogniK5377) had attempted to implement NVDEC before and reached out to get an understanding of this feature's scope and challenges faced along the way. 
+Before I knew it, I had access to the repository with the old implementation's code and was encouraged to find a way to finish it off.
+
+It was intimidating at first, but the encouragement and mentorship from ogniK kept me motivated. 
+It was also the first glimpse I had at the guts of yuzu, and found the work that has gone into the emulation of the Nintendo Switch's GPU to be very interesting.
+
+{{< single-title-imgs
+    "The Legend of Zelda: Link's Awakening"
+    "./zla_0.png"
+    "./zla_1.png" 
+>}}
+
+{{< single-title-imgs
+    "Super Smash Bros. Ultimate"
+    "./ssbu_0.png"
+    "./ssbu_1.png" 
+>}}
+
+**Q: What obstacles did you have to overcome when implementing NVDEC?**
+
+**E:** The earliest challenge was to understand what the data I'm looking at means. At the GPU command level, everything is hexadecimal numbers with no obvious meaning tied to it. 
+Thankfully, there have been others who reverse engineered and documented the processes that the GPU goes through for the NVDEC functionality. 
+I was able to get quite far with decoding H.264 videos with the help of ogniK's older implementation. But the VP9 encoded videos found in many games were very difficult to decode. 
+
+In order to decode VP9, I needed the frame data along with a "compressed" and "uncompressed" header to be sent to [FFmpeg](https://ffmpeg.org/), a library which is widely used for video decoding. 
+The Switch GPU provided me with the frame data, but the headers needed to be manually constructed based on meta data provided by the GPU. 
+I found myself spending countless hours reading the VP9 specification, and analyzing the video frame-by-frame in a tool that displays each frame's headers. Ultimately, it paid off, and many VP9 videos are decoded accurately.
+
+{{< imgs
+    "./ssbu_2.mp4| Footage of Super Smash Bros. Ultimate's intro video early in development."
```

Comment:
The autoplaying video is very distracting while reading the article (due to the flickering). Since we don't know how to not auto-play videos, a workaround could be to upload unlisted to youtube and then embed the video. Youtube videos won't autoplay